swift_package/BrainFlowCI/BrainFlowBindings/BoardShim.swift

```diff
@@ -0,0 +1,582 @@
+//
+//  BoardShim.swift
+//  a Swift binding for BrainFlow's board_shim high-level API
+//
+//  Created by Scott Miller for Aeris Rising, LLC on 8/23/21.
+//
+
+import Foundation
+
+struct BoardShim {
+    let boardId: BoardIds
+    let masterBoardId: BoardIds
+    let bfParams: BrainFlowInputParams
+    private let jsonBrainFlowInputParams: [CChar]
+    
+    /*
+    try:
+        self.input_json = input_params.to_json().encode()
+    except BaseException:
+        self.input_json = input_params.to_json()
+    self.board_id = board_id
+    # we need it for streaming board
+    if board_id == BoardIds.STREAMING_BOARD.value or board_id == BoardIds.PLAYBACK_FILE_BOARD.value:
+        if input_params.master_board != BoardIds.NO_BOARD:
+            self._master_board_id = input_params.master_board
+        else:
+            raise BrainFlowError('you need set master board id in BrainFlowInputParams',
+                                 BrainFlowExitCodes.INVALID_ARGUMENTS_ERROR.value)
+    else:
+        self._master_board_id = self.board_id
+    */
+        
+    init (_ boardId: BoardIds, _ params: BrainFlowInputParams) throws {
+        self.boardId = boardId
+        self.bfParams = params
+        self.masterBoardId = try BoardShim.getMasterBoardID(boardId: boardId, params: params)
```

Comment:
this method is called getBoardID in all other bindings

Reply:
I have both getBoardID and getMasterBoardID. I can make getMasterBoardID a private func, so that it is not available outside the bindings. Will that work?

Reply:
getBoardId should return master board id for streaming and playback and normal board id for everything else. Having both is confusing

Reply:
But if its used only inside binding its probably ok